### DISCUSSION TITLE
CRC vs OCP - is microshift the way to be able to develop locally?

### DISCUSSION THREAD

**Discussion by darkedges** [crc-org/crc#3655] [Q&A]
Afternoon,
Client has OCP deployed and I am trying to work out how to best support local development using CRC. If I am understanding this correctly CRC deploys services on top of OCP to provide the UI. If I want to be able to enable local development i.e use tooling like `DevSpace` that can provide Container and Kustomize builds I am thinking I can just deploy `microshift` to give them an equivalent environment. 

Or have I misunderstood what `microshift` is and need `openshift` to provide a similar deployment environment?

**Comment by praveenkumar** [comment]
I think your question is mostly around `microshift` vs `openshift` and you want to know if some deployments which works on OCP side can work on `microshift` also? https://github.com/openshift/microshift/blob/main/docs/design.md#design-goals have better description around it.

**Comment by praveenkumar** [comment]
Also check https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/getting_started/microshift-architecture#microshift-differences-oke_microshift-architecture one

**Comment by cfergeau** [comment]
> If I am understanding this correctly CRC deploys services on top of OCP to provide the UI.

Just a clarification, CRC+openshift runs a full OCP cluster inside a VM, and it does not add services on top of OCP. You can use your usual workloads on it for experiments, but the drawback is that in this configuration, resource usage is quite high.
If you can deploy your workloads on microshift, then yes, it may be the way to go for your usecase.
